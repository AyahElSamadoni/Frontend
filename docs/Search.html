<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Search</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Search</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Search<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">This is a description of the Search Component.</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="Search"><span class="type-signature"></span>new Search<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Containers_Search_Search.js.html">Containers/Search/Search.js</a>, <a href="Containers_Search_Search.js.html#line20">line 20</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="state"><span class="type-signature"></span>state<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>resultsArr</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;json></span>


            
            </td>

            

            

            <td class="description last">Array of the search results.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>loading</code></td>
            

            <td class="type">
            
                
<span class="param-type">bool</span>


            
            </td>

            

            

            <td class="description last">Indicates whether the component is still loading or ready.</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Containers_Search_Search.js.html">Containers/Search/Search.js</a>, <a href="Containers_Search_Search.js.html#line30">line 30</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="editProfile.html">editProfile</a></li><li><a href="forgotPassword.html">forgotPassword</a></li><li><a href="login.html">login</a></li><li><a href="Newsfeed.html">Newsfeed</a></li><li><a href="profile.html">profile</a></li><li><a href="Search.html">Search</a></li><li><a href="signup.html">signup</a></li></ul><h3>Global</h3><ul><li><a href="global.html#button">button</a></li><li><a href="global.html#input">input</a></li><li><a href="global.html#LandingPage">LandingPage</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Tue May 07 2019 23:51:49 GMT+0200 (Egypt Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>