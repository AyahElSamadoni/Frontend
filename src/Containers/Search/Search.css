.scroll-enter {
  opacity: 0.01;
  transform: translate(0, -500);
}

.scroll-enter-active {
  opacity: 1;
  transform: translate(0, 0);
  transition: all 500ms ease-in;
}

.scroll-exit {
  opacity: 1;
  transform: translate(0, 0);
}

.scroll-exit-active {
  opacity: 0.01;
  transform: translate(100px, 0);
  transition: all 500ms ease-in;
}
