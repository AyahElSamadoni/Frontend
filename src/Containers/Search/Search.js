import React, { Component } from "react";
import Nav from "../../Components/NavBar/NavBar";
import ProfileSearch from "../../Components/profileSearch/profileSearch";
import Axios from "../../axios-users";
import { connect } from "react-redux";
import { zoomInUp } from "react-animations";
import Spinner from "../../Components/UI/Spinner/Spinner";
import "./Search.css";
import {
  CSSTransition,
  Transition,
  TransitionGroup
} from "react-transition-group";
import { fabomb, faBomb } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
/**
 * This is a description of the Search Component.
 * @class
 * */
class Search extends Component {
  constructor(props) {
    super(props);
    /**
    * @property {json[]} resultsArr -Array of the search results.
   
    * @property {bool} loading                -Indicates whether the component is still loading or ready.
   
    
  */
    this.state = {
      resultsArr: [],
      loading: true,
      results: null
    };
  }
  // componentDidMount() {
  //   Axios.get(
  //     "http://localhost:8080/users/search?query=" +
  //       this.props.match.params.value,
  //     {
  //       headers: {
  //         token: localStorage.getItem("jwtToken")
  //       }
  //     }
  //   )
  //     .then(res => {
  //       console.log("success from search");
  //       console.log(res.data);
  //     })
  //     .catch(err => {
  //       console.log("failure from search", { ...err });
  //     });
  // }
  // componentWillUpdate() {
  //   Axios.get(
  //     "http://localhost:8080/users/search?query=" +
  //       this.props.match.params.value,
  //     {
  //       headers: {
  //         token: localStorage.getItem("jwtToken")
  //       }
  //     }
  //   )
  //     .then(res => {
  //       console.log("success from search");
  //       console.log(res.data);
  //     })
  //     .catch(err => {
  //       console.log("failure from search", { ...err });
  //     });
  // }
  async componentWillReceiveProps(nextProps) {
    this.setState({ loading: true });
    await Axios.get("/users/search?query=" + nextProps.match.params.value, {
      headers: {
        token: localStorage.getItem("jwtToken")
      }
    })
      .then(async res => {
        console.log("success from search");
        console.log(res.data);
        await this.setState({ resultsArr: res.data });
        const showResults = this.state.resultsArr.map(profile => {
          console.log("auth:");
          console.log(this.props.auth);
          return (
            <CSSTransition key={profile._id} timeout={500} classNames="scroll">
              <button
                type="button"
                className="list-group-item list-group-item-action"
              >
                <ProfileSearch
                  key={profile._id}
                  name={profile.name}
                  username={profile.screen_name}
                  text={profile.bio}
                  profile_image_url={profile.profile_image_url}
                />
              </button>
            </CSSTransition>
          );
        });
        this.setState({ results: showResults });
      })
      .catch(err => {
        console.log("failure from search", { ...err });
      });
    this.setState({ loading: false });
  }
  async componentDidMount() {
    this.setState({ loading: true });
    await Axios.get("/users/search?query=" + this.props.match.params.value, {
      headers: {
        token: localStorage.getItem("jwtToken")
      }
    })
      .then(async res => {
        console.log("success from search");
        console.log(res.data);
        await this.setState({ resultsArr: res.data });
        const showResults = this.state.resultsArr.map(profile => {
          console.log("auth:");
          console.log(this.props.auth);
          return (
            <CSSTransition key={profile._id} timeout={500} classNames="scroll">
              <button
                type="button"
                className="list-group-item list-group-item-action"
              >
                <ProfileSearch
                  key={profile._id}
                  name={profile.name}
                  username={profile.screen_name}
                  text={profile.bio}
                  profile_image_url={profile.profile_image_url}
                />
              </button>
            </CSSTransition>
          );
        });
        this.setState({ results: showResults });
      })
      .catch(err => {
        console.log("failure from search", { ...err });
      });
    this.setState({ loading: false });
  }
  render() {
    // this.getSearchResults();z
    return (
      <div className="container-fluid">
        <div style={{ marginBottom: "1%" }}>
          <Nav
            notifcationsCount={this.props.notifications.notifications.length}
          />
        </div>
        <div className="container" style={{ width: "80%" }}>
          {this.state.loading ? (
            <div
              class="d-flex justify-content-center"
              style={{ marginTop: "2%" }}
            >
              <Spinner />
            </div>
          ) : (
            <div>
              {this.state.resultsArr.length > 0 ? (
                <div className="list-group">
                  <button className="list-group-item list-group-item-action active">
                    Search Results
                  </button>
                  <TransitionGroup>{this.state.results}</TransitionGroup>
                </div>
              ) : (
                <div style={{ textAlign: "center", marginTop: "10%" }}>
                  <FontAwesomeIcon
                    icon={faBomb}
                    size={"10x"}
                    style={{ color: "black", textAlign: "center" }}
                  />

                  <h1 style={{ textAlign: "center", marginTop: "2%" }}>
                    No results
                  </h1>
                </div>
              )}
            </div>
          )}
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  auth: state.auth,
  notifications: state.notifications
});

export default connect(mapStateToProps)(Search);
