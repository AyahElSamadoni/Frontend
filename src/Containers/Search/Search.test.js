import React from "react";
import { shallow, configure } from "enzyme";
import Search from "./Search";
import Nav from "../../Components/NavBar/NavBar";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import ProfileSearch from "../../Components/profileSearch/profileSearch";
import {
  CSSTransition,
  Transition,
  TransitionGroup
} from "react-transition-group";
configure({ adapter: new Adapter() });
describe("<editProfile>", () => {
  it("should be defined", () => {
    expect(<Search />).toBeDefined();
  });
  it("should render", () => {
    const wrapper = shallow(<Search />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should render", () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });
  it("check that props are send correctly to profile search", () => {
    //const wrapper = shallow(<Nav />);
    const props = {
      key: "123",
      name: "mirna",
      username: "mirna",
      text: "ana mirna",
      profile_image_url: "hhttp:///123354"
    };
    const wrapper = shallow(<ProfileSearch {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("check that props are send correctly to css transition", () => {
    //const wrapper = shallow(<Nav />);
    const props = {
      key: "123",
      timeout: "500",
      classNames: "scroll"
    };
    const wrapper = shallow(<CSSTransition {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
