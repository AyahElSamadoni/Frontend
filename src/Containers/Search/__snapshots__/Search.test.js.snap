// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<editProfile> check that props are send correctly to profile search 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileSearch
    name="mirna"
    profile_image_url="hhttp:///123354"
    text="ana mirna"
    username="mirna"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row justify-content-start"
      >
        <div
          className="col-md-auto "
        >
          <img
            className="img-fluid"
            src="hhttp:///123354"
            style={
              Object {
                "width": "75px",
              }
            }
          />
        </div>
        <div
          className="col-md-auto"
        >
          <div
            className="row"
          >
             
            mirna
             
          </div>
          <div
            className="row"
          >
             @
             
          </div>
          <div
            className="row"
          >
            <p>
              ana mirna
            </p>
          </div>
        </div>
        <div
          className="col-1 ml-auto"
          style={
            Object {
              "marginTop": "0.5%",
            }
          }
        >
          <button
            className="btn btn-primary btn-sm"
            type="button"
          >
            Unfollow
          </button>
        </div>
      </div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-md-auto "
          >
            <img
              className="img-fluid"
              src="hhttp:///123354"
              style={
                Object {
                  "width": "75px",
                }
              }
            />
          </div>,
          <div
            className="col-md-auto"
          >
            <div
              className="row"
            >
               
              mirna
               
            </div>
            <div
              className="row"
            >
               @
               
            </div>
            <div
              className="row"
            >
              <p>
                ana mirna
              </p>
            </div>
          </div>,
          <div
            className="col-1 ml-auto"
            style={
              Object {
                "marginTop": "0.5%",
              }
            }
          >
            <button
              className="btn btn-primary btn-sm"
              type="button"
            >
              Unfollow
            </button>
          </div>,
        ],
        "className": "row justify-content-start",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="img-fluid"
              src="hhttp:///123354"
              style={
                Object {
                  "width": "75px",
                }
              }
            />,
            "className": "col-md-auto ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "img-fluid",
              "src": "hhttp:///123354",
              "style": Object {
                "width": "75px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="row"
              >
                 
                mirna
                 
              </div>,
              <div
                className="row"
              >
                 @
                 
              </div>,
              <div
                className="row"
              >
                <p>
                  ana mirna
                </p>
              </div>,
            ],
            "className": "col-md-auto",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "mirna",
                  " ",
                ],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "mirna",
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " @",
                  undefined,
                  " ",
                ],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [
                " @",
                undefined,
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  ana mirna
                </p>,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ana mirna",
                },
                "ref": null,
                "rendered": "ana mirna",
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-primary btn-sm"
              type="button"
            >
              Unfollow
            </button>,
            "className": "col-1 ml-auto",
            "style": Object {
              "marginTop": "0.5%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Unfollow",
              "className": "btn btn-primary btn-sm",
              "type": "button",
            },
            "ref": null,
            "rendered": "Unfollow",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row justify-content-start"
        >
          <div
            className="col-md-auto "
          >
            <img
              className="img-fluid"
              src="hhttp:///123354"
              style={
                Object {
                  "width": "75px",
                }
              }
            />
          </div>
          <div
            className="col-md-auto"
          >
            <div
              className="row"
            >
               
              mirna
               
            </div>
            <div
              className="row"
            >
               @
               
            </div>
            <div
              className="row"
            >
              <p>
                ana mirna
              </p>
            </div>
          </div>
          <div
            className="col-1 ml-auto"
            style={
              Object {
                "marginTop": "0.5%",
              }
            }
          >
            <button
              className="btn btn-primary btn-sm"
              type="button"
            >
              Unfollow
            </button>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-md-auto "
            >
              <img
                className="img-fluid"
                src="hhttp:///123354"
                style={
                  Object {
                    "width": "75px",
                  }
                }
              />
            </div>,
            <div
              className="col-md-auto"
            >
              <div
                className="row"
              >
                 
                mirna
                 
              </div>
              <div
                className="row"
              >
                 @
                 
              </div>
              <div
                className="row"
              >
                <p>
                  ana mirna
                </p>
              </div>
            </div>,
            <div
              className="col-1 ml-auto"
              style={
                Object {
                  "marginTop": "0.5%",
                }
              }
            >
              <button
                className="btn btn-primary btn-sm"
                type="button"
              >
                Unfollow
              </button>
            </div>,
          ],
          "className": "row justify-content-start",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="img-fluid"
                src="hhttp:///123354"
                style={
                  Object {
                    "width": "75px",
                  }
                }
              />,
              "className": "col-md-auto ",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "img-fluid",
                "src": "hhttp:///123354",
                "style": Object {
                  "width": "75px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="row"
                >
                   
                  mirna
                   
                </div>,
                <div
                  className="row"
                >
                   @
                   
                </div>,
                <div
                  className="row"
                >
                  <p>
                    ana mirna
                  </p>
                </div>,
              ],
              "className": "col-md-auto",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "mirna",
                    " ",
                  ],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "mirna",
                  " ",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " @",
                    undefined,
                    " ",
                  ],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [
                  " @",
                  undefined,
                  " ",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    ana mirna
                  </p>,
                  "className": "row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ana mirna",
                  },
                  "ref": null,
                  "rendered": "ana mirna",
                  "type": "p",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-primary btn-sm"
                type="button"
              >
                Unfollow
              </button>,
              "className": "col-1 ml-auto",
              "style": Object {
                "marginTop": "0.5%",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Unfollow",
                "className": "btn btn-primary btn-sm",
                "type": "button",
              },
              "ref": null,
              "rendered": "Unfollow",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<editProfile> check the type of value 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CSSTransition
    classNames="scroll"
    timeout="500"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appear": false,
      "enter": true,
      "exit": true,
      "in": false,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "timeout": "500",
      "unmountOnExit": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appear": false,
        "enter": true,
        "exit": true,
        "in": false,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "timeout": "500",
        "unmountOnExit": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`<editProfile> should render 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<editProfile> should render 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Search) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
